### ISSUE TITLE
browse using arrow keys doesn't always work

### ISSUE THREAD

**Issue by mike-wiese** [a2-4am/4cade#303]
I'm using rev 04a1, on my Apple II+ and IIgs.

At the main screen, Type X which takes you to Xevious. Then type right arrow which takes you to Zargs. Then type left arrow and you are taken back to Xevious. This makes sense.

However if you start from the main screen and Type Z which takes you to Zargs, and then left arrow you are returned to the main screen. I expected to be taken to the previous game in the list, i.e. Xevious.

A related complaint (and I know this is subjective) is that the name matching via the keyboard is unintuitive. For example, when I type D from the main screen I would expect to be taken to the first game with a name that starts with "D". Instead I am taken to "Drol". In addition, sometimes when I type more chars the result is a beep. I guess this means no match, and it appears I can type ESC to clear the name matching buffer and start over. But this is not explained in the help.

Since the string that is used for matching is invisible, it's kinda hard to follow. I think it should work like type selecting a list of files in the Finder and open/save dialogs on macOS, since people are more likely to be used to that behavior. When you type a char it selects the first file name that starts with the char (or if none match, the first file that comes after the char typed). If you keep typing within a short timeout it appends to the search string to refine the search. After the timeout it no longer appends, instead it is treated as a new search.

**Comment by a2-4am** [comment]
So there are multiple issues here. The left-arrow-should-browse-left issue is a subtle interaction between search and browse modes, and the current behavior is the best compromise we could implement given all the constraints, most of which are non-obvious. The ][+ (which is supported hardware) has no dedicated Delete key, so left-arrow needs to be synonymous with Delete to back up your input. Some emulators also "helpfully" map Delete to left-arrow at a low level, so Delete needs to be synonymous with left-arrow even on non-][+ platforms.

The name matching is not designed to match the Finder; it is designed to find the "best" match in the fewest number of keystrokes. So "D" matches "Drol" because the "D" is the first letter (given more weight than subsequent letters) and because "Drol" is the shortest overall match. So "W N" will match "Wavy Navy" because first letters of words are given more weight. "N M P" will match "Night Mission Pinball", &c.

There is a visual indicator of the input buffer: the dots above letters show which letters were matched.

In short, the way the UI works is as close to my vision of how it ought to work, given the constraints. I'm sorry you find it confusing, but we will not be changing it.
